dpc-testing/performance/pkg/dpc/generator.go

```diff
@@ -0,0 +1,99 @@
+package dpc
+
+import (
+	"bytes"
+	"crypto/rsa"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+)
+
+func templateBodyGenerator(templateFile string, replacementMap map[string]func() string) func() []byte {
+	body, err := ioutil.ReadFile(templateFile)
+	if err != nil {
+		panic(err)
+	}
+	return func() []byte {
+		output := body
+
+		for k, v := range replacementMap {
+			keyCount := bytes.Count(body, []byte(k))
+			for i := 0; i < keyCount; i++ {
+				value := v()
+				output = bytes.Replace(output, []byte(k), []byte(value), i+1)
+			}
+
+		}
+
+		return output
+	}
+}
+
+func keyBodyGenerator(n int, fn func() (string, *rsa.PrivateKey, string)) func() []byte {
+	var bodies [][]byte
+	for i := 0; i < n; i++ {
+		pubKeyStr, _, signature := fn()
+		bodies = append(bodies, []byte(fmt.Sprintf("{ \"key\": \"%s\", \"signature\": \"%s\"}", pubKeyStr, signature)))
```

Comment:
This will be more readable with `` instead of "":
```suggestion
		bodies = append(bodies, []byte(fmt.Sprintf(`{ "key": "%s", "signature": "%s"}`, pubKeyStr, signature)))
```